product_manager.py

```diff
@@ -40,3 +45,12 @@ def get_all_matching_products(self, search_word):
             item_details.update({'Tags': item_tag_list})
             all_maching_products.append(item_details)
         return all_maching_products
+    
+    def get_product_object(self, item_id: str):
+        item_details = self.item_manager.get_item_details(item_id)
```

Comment:
You probably want an ItemDetails dataclass return by get_item_details. But perhaps refactor in another commit..

Reply:
possibly, let's discuss later for potential implementation in separate work stream